politeiawww/codetracker/github/api/types.go

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) 2020 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package api
+
+// Author has basic information about the creator of a commit.
+type Author struct {
+	Name  string `json:"name"`
+	Email string `json:"email"`
+	Date  string `json:"date"`
+}
+
+// Commit holds basic information about a given commit, but no details
+// about the actual changes.
+type Commit struct {
+	Author    Author `json:"author"`
+	Committer Author `json:"committer"`
+	Message   string `json:"message"`
+}
+
+// CommitParent has information about a commit's parent commit.
+type CommitParent struct {
+	SHA string `json:"sha"`
+	URL string `json:"url"`
+}
+
+// CommitStats contains basic LOC information about a given commit.
+type CommitStats struct {
+	Additions int `json:"additions"`
+	Deletions int `json:"deletions"`
+	Total     int `json:"total"`
+}
+
+type PullRequestCommit struct {
+	SHA       string         `json:"sha"`
+	URL       string         `json:"url"`
+	Parents   []CommitParent `json:"parents"`
+	Stats     CommitStats    `json:"stats"`
+	Commit    Commit         `json:"commit"`
+	Author    User           `json:"author"`
+	Committer User           `json:"committer"`
+
+	// local change
+	Discarded bool `json:"discarded"`
+}
+
+// PullsRequest contains all high level information returned from the
+// PullsRequest request.
+type PullsRequest struct {
+	URL            string `json:"url"`
+	Number         int    `json:"number"`
+	State          string `json:"state"`
+	Title          string `json:"title"`
+	User           User   `json:"user"`
+	UpdatedAt      string `json:"updated_at"`
```

Comment:
Can we make this a proper json formated or this dictated by github?

If this is not dictated please remove the `_`.

Reply:
Yah these are dictated by github  so needed for proper decoding:
https://developer.github.com/v3/pulls/